### PR TITLE
Initial thoughts on an introductory screencast

### PR DESCRIPTION
**Pull request by parkr**
Docs sites are so 2013. Let's start making short, simple screencasts around each feature in Jekyll to facilitate Jekyll's use. Potential topics:
- Posts
- Pages
- Pagination
- Markdown converters
- Data files
- Plugins
- Eventually Sass, CoffeeScript & Collections

_etc._

/cc @jekyll/outreach @troyswanson

### CODE REVIEW

File: 01-introduction.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+- Easy to install: `gem install ruby`
+- Can write your content (posts & pages) in HTML, Markdown, or Textile
+- Specify data about the posts & pages in YAML at the top of the file, called "YAML Front-Matter"
+- Built-in pagination support for posts
+- Draft workflow provided but optional
+- Uses Liquid templating language for semi-dynamic pages (change on each build)
+- Supports partials ("includes") and layouts to DRY-up your code
+- Can contain arbitrary data as YAML, in `_data` folder
+- Customizable with plugins
+- Can host your site for free on GitHub Pages
```
Comment (parkr) [inline 01-introduction.md:19]: Might want to list GHP with Heroku, OpenShift, etc. as alternatives

Comment (mattr-) [inline 01-introduction.md:19]: How long do we want these to be? What formats would we publish them in? How/where would we host them?

/me goes to reinstall his copy of ScreenFlow

**Comment by parkr** [comment]
@mattr-:

> How long do we want these to be?

2-3 minutes? Depends on how in-depth we want to go. Everything about Jekyll is simple by example.

> What formats would we publish them in?

I'd say upload them to YouTube or Vimeo and not worry about the formats.

> How/where would we host them?

YouTube or Vimeo would host the videos, but then I'd love to make a quick Jekyll site which collects them all in a Jekyll-branded way.

> /me goes to reinstall his copy of ScreenFlow

:smiley: :+1: :100: :factory:

File: 01-introduction.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# 01 - Introduction
+
+## An introduction to Jekyll
+
+- Jekyll is a blog-aware static site generator
+- Originally created by Tom Preston-Werner (mojombo), President of GitHub
+- Entirely open-source: github.com/jekyll/jekyll
+- It uses only files - no databases needed!
+- Written entirely in Ruby, tested on Linux & Mac OS X, generally works on Windows
+- Easy to install: `gem install ruby`
```
Comment (tombell) [inline 01-introduction.md:10]: wat.

**Comment by mattr-** [comment]
I think the intro screencast needs to be split up. I'm uncertain that all those things could be covered and adequately demonstrated in two to three minutes.

**Comment by troyswanson** [comment]
Have you seen [Asciinema](http://asciinema.org/)? Might be a cool way to demonstrate command line stuff, although I imagine the less time spent in Terminal, the better.

File: 01-introduction.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# 01 - Introduction
+
+## An introduction to Jekyll
+
+- Jekyll is a blog-aware static site generator
+- Originally created by Tom Preston-Werner (mojombo), President of GitHub
+- Entirely open-source: github.com/jekyll/jekyll
+- It uses only files - no databases needed!
```
Comment (troyswanson) [inline 01-introduction.md:8]: More than just "no databases" - no interpreter needed on page loads. Emphasize the static pages that are rendered and hit home that any dumb HTTP server will work, sans php or anything else.

File: 01-introduction.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+- Built-in pagination support for posts
+- Draft workflow provided but optional
+- Uses Liquid templating language for semi-dynamic pages (change on each build)
+- Supports partials ("includes") and layouts to DRY-up your code
+- Can contain arbitrary data as YAML, in `_data` folder
+- Customizable with plugins
+- Can host your site for free on GitHub Pages
+- `jekyll-import` allows for super easy importing of data from other sources
+
+Maybe we should also offer a survey to get more info about where new users are coming from?
```
Comment (troyswanson) [inline 01-introduction.md:22]: Indeed, but be sure to phrase the questions in a way that prompts respondents to remember back to when they were first learning about Jekyll. If I'm reading this correctly, we want this to be one of the first ways a newbie can learn about the project.

**Comment by troyswanson** [comment]
Maybe before the YAML Front-Matter and the `_data` tutorials, we could do a crash course on YAML in general. I know when I got started, I had no idea about YAML or its syntax.

File: 01-introduction.md (hunk 2 above)
Comment (parkr) [inline 01-introduction.md:10]: :laughing:

**Comment by parkr** [comment]
> I think the intro screencast needs to be split up. I'm uncertain that all those things could be covered and adequately demonstrated in two to three minutes.

@mattr- The intro screencast wouldn't aim to adequately demonstrate everything, it should introduce Jekyll, how it works at a very high level, why you might want to use it, and some examples of awesomeness done by Jekyll. And I'm happy to set the limit to 5 minutes, but we should shoot for around 3 minutes.

**Comment by parkr** [comment]
> Maybe before the YAML Front-Matter and the _data tutorials, we could do a crash course on YAML in general. I know when I got started, I had no idea about YAML or its syntax.

@troyswanson Great idea! Open a PR? Would be great to have your input for that.

**Comment by Wolfr** [comment]
Just wanted to chime in and let you know that I have a mic setup and screencasting software to do some basic screencasts on Jekyll so I can definitely help, just ask me.

I have a real business need for this so I am motivated to work on this. I am using Jekyll as part of my toolbelt to teach HTML prototyping to not-so-tech folks used to working in Balsamiq/Axure.

Here is a presentation from last year: http://www.youtube.com/watch?v=O7NBEFmA7yA&feature=c4-overview&list=UUMs4dC8I8MqX-ZbfmSi0MAQ . It was part of a mini Jekyll meetup (15ppl) I organized.

It's kind of long and I mumble a lot but I can imagine if I make some screencasts with a few takes at home, each 3-5 mins about a single subject, speaking clearly into my USB mic + Youtube subtitles it will be very understandable.

File: 01-introduction.md (hunk 4 above)
Comment (parkr) [inline 01-introduction.md:22]: > If I'm reading this correctly, we want this to be one of the first ways a newbie can learn about the project.

:+1: Yes! Exactly that.

Want to take a stab at a user survey back in jekyll/jekyll? I'll convert it to PollDaddy and tweet it out.

**Comment by troyswanson** [comment]
I put together a basic survey question that will hopefully help us figure out what the most pressing pain points folks have when dealing with Jekyll. Take a gander over at jekyll/jekyll#1994.

File: 01-introduction.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
+# 01 - Introduction
+
+## An introduction to Jekyll
+
+- Jekyll is a blog-aware static site generator
+- Originally created by Tom Preston-Werner (mojombo), President of GitHub
+- Entirely open-source: github.com/jekyll/jekyll
```
Comment (zlatanvasovic) [inline 01-introduction.md:7]: Can you add `https://` to the start of URL? That will _linkify_ the _link_.

Comment (parkr) [inline 01-introduction.md:7]: This file is just notes for what will eventually turn into a video. No reason to linkify this.

Comment (zlatanvasovic) [inline 01-introduction.md:7]: Oh, good then! :)

**Comment by doktorbro** [comment]
> Docs sites are so 2013. Let's start making short, simple screencasts around each feature in Jekyll to facilitate Jekyll's use.

Keep in mind that non-English speakers always prefer _written_ words. People like me have only a limited receptivity for spoken words, so we can not really benefit from screencasts.

Moreover companies often block Youtube in their networks, so screencasts will be unaccessible for big business at work.

**Comment by zlatanvasovic** [comment]
I agree.

**Comment by troyswanson** [comment]
Yeah, but that shouldn't stop us from doing it..

**Comment by ndarville** [comment]
What would be the right way to support users on both Mac/UNIX and Windows?
- Include both in one video
- Split them into separate videos

The latter might only work on YouTube where you can include an annotation to work as a link to another video.

I’d also say that the current text is aimed at an intermedia technical audience. Not that it’s a bad thing; we just have to decide whom we assume are in the audience.

I support the idea as well; I was just about to suggest it in fact.

**Comment by nicksuch** [comment]
For reference, I have a few Jekyll screencasts up on Udemy. I was still
pretty new to Jekyll (and screencasting) when I made this, so it's not
great, but possibly a starting point. Here's the link:
https://www.udemy.com/jekyll-and-github-pages/?couponCode=jekyllscreencastThe
coupon code "jekyllscreencast" should get you past the paywall, let me
know if it gives you any issues.

-Nick

---

nick such
502.802.5972 | nicksuch@gmail.com
nicksuch.com | @aerosuch http://twitter.com/aerosuch |
meet?http://doodle.com/nicksuch

#LearnToCode -> _AwesomeIncU.com_ http://awesomeincu.com/

**Comment by justinmusgrove** [comment]
I am new to screen casting and no way near @ryanb material but learning and having fun. I put together a rough cut quick start and pushed it to youtube https://www.youtube.com/watch?v=jp_4SHiE6_I. I like the three minute limit as it will force you to break down material in very focus manner which in turn will answer users questions and give enough context/confidence to explore. I too agree that each version should contain a transcript for all users to read/translate. In regards to branding, you could host it up on youtube and use http://www.sublimevideo.net/ to give it a different feel on screencasts.jekyllrb.com.

**Pull request closed without merging**